### PR TITLE
CAMS-525: Implement Default Divisions on Search Screen

### PR DESCRIPTION
**Pull request by fmaddenflx**
# Purpose

Implement simple changes to ComboBox rather than reimplementing the entire thing.  Complete 525 such that the users default divisions show up at the top of the combo box list by default and user can then change their selections if need be.

# Major Changes

Major changes to ComboBox (though not as drastic as our first attempts).

# Testing/Validation

Tests have all been updated.

# Notes

A lot of updates were made to combo box.  look 'em over.

# Definition of Done:

- [ ] Code refactored for clarity - Developers can understand the work simply by reviewing the code
- [ ] Dependency rule followed - More important code doesn’t directly depend on less important code
- [ ] Development debt eliminated - UX and code aligns to the team’s latest understanding of the domain

## Summary by Sourcery

Refactor the ComboBox component to simplify its implementation and improve user experience, focusing on default division selection and accessibility

New Features:
- Implement default division selection at the top of the ComboBox list
- Add more descriptive aria labels for ComboBox items
- Improve selection and filtering behavior

Enhancements:
- Simplified ComboBox state management
- Improved keyboard navigation
- Enhanced selection and clear functionality
- More consistent focus management

Tests:
- Updated test cases to reflect new ComboBox behavior
- Added comprehensive aria label test cases
- Improved test coverage for keyboard interactions

### CODE REVIEW

File: user-interface/src/admin/privileged-identity/PrivilegedIdentity.tsx [hunk 1]
```diff
@@ -113,9 +112,9 @@ export function PrivilegedIdentity() {
   }
 
   function handleSelectUser(options: ComboOption[]) {
-    if (options?.length === 1) {
+    if (options.length === 1) {
       const userId = options[0].value;
-      setSelectedUser(userList.find((user) => user.id === options[0].value)!);
+      // setSelectedUser(userList.find((user) => user.id === options[0].value)!);
```
Comment (fmaddenflx) [inline user-interface/src/admin/privileged-identity/PrivilegedIdentity.tsx:117]: Um.  So this method is called handleSelectUser().  So why did we comment out setSelectedUser?  is that no longer a state variable?

Comment (btposey) [inline user-interface/src/admin/privileged-identity/PrivilegedIdentity.tsx:117]:
```suggestion
```

Comment (jamesobrooks) [inline user-interface/src/admin/privileged-identity/PrivilegedIdentity.tsx:117]: Remove if not needed.

File: user-interface/src/case-detail/CaseDetailScreen.tsx [hunk 2]
```diff
@@ -592,12 +592,13 @@ export default function CaseDetailScreen(props: CaseDetailProps) {
...
                         options={getSummaryFacetList(caseDocketSummaryFacets)}
                         onClose={handleSelectedFacet}
-                        onPillSelection={handleSelectedFacet}
                         onUpdateSelection={handleFacetClear}
                         label="Filter by Summary"
                         ariaDescription="Select multiple options. Results will update when the dropdown is closed."
                         aria-live="off"
                         multiSelect={true}
+                        singularLabel="summary"
+                        pluralLabel="summaries"
```
Comment (jamesobrooks) [inline user-interface/src/case-detail/CaseDetailScreen.tsx:601]: Oh, here's a counter example for my earlier claim that we probably wouldn't be encountering edge cases. 😁

File: user-interface/src/lib/components/combobox/ComboBox.test.tsx [hunk 3]
```diff
@@ -326,31 +408,37 @@ describe('test cams combobox', () => {
...
+    // await waitFor(() => {
+    //  expect(document.querySelector('.input1')).toHaveFocus();
+    // });
   });
 
-  test('If the dropdown list is closed, then typing characters on the keyboard should open the dropdown list', async () => {
-    renderWithProps();
+  test('should not open when combobox is disabled', async () => {
+    renderWithProps({ disabled: true });
+    // await toggleDropdown();
```
Comment (En-8) [inline user-interface/src/lib/components/combobox/ComboBox.test.tsx:435]: Remove?

File: user-interface/src/lib/components/combobox/ComboBox.test.tsx [hunk 4]
```diff
@@ -326,31 +408,37 @@ describe('test cams combobox', () => {
...
     });
+    const result = ref.current?.getSelections();
+    expect(result).toEqual([]);
+
     expect(isDropdownClosed()).toBeTruthy();
+
+    // TODO: This works manually.  Figure out why this focus isn't happening.
+    // await waitFor(() => {
+    //  expect(document.querySelector('.input1')).toHaveFocus();
+    // });
```
Comment (En-8) [inline user-interface/src/lib/components/combobox/ComboBox.test.tsx:430]: Are we leaving this as a "later" thing or should we look into it as part of this PR?

File: user-interface/src/lib/components/combobox/ComboBox.test.tsx [hunk 5]
```diff
@@ -856,36 +793,30 @@ describe('test cams combobox', () => {
...
-
-    await waitFor(() => {
-      const input = document.querySelector('.combo-box-input');
-      expect(input).toBeInTheDocument();
-      expect(input).toBeVisible();
-    });
-
     ref.current?.disable(true);
 
+    // const expandButton = document.querySelector('.expand-button');
```
Comment (En-8) [inline user-interface/src/lib/components/combobox/ComboBox.test.tsx:807]: Remove?

File: user-interface/src/data-verification/transfer/SuggestedTransferCases.tsx [hunk 6]
```diff
@@ -156,7 +156,7 @@ function _SuggestedTransferCases(
     // TODO: Make sure the following only happens when we click the 'Clear' button, not the 'go back' button on the modal
```
Comment (jamesobrooks) [inline user-interface/src/data-verification/transfer/SuggestedTransferCases.tsx:156]: This seems like an easily verified TODO.

File: user-interface/src/lib/testing/testing-utilities.ts [hunk 7]
```diff
@@ -111,7 +111,9 @@ function selectRadio(id: string) {
   return radio;
 }
 
-async function selectComboBoxItem(id: string, itemIndex: number = 0) {
+// TODO: either turn this into a toggle or create an unselect function
```
Comment (En-8) [inline user-interface/src/lib/testing/testing-utilities.ts:114]: Should we do this?

File: user-interface/src/search/SearchScreen.tsx [hunk 8]
```diff
@@ -131,6 +147,7 @@ export default function SearchScreen() {
   function handleChapterSelection(selections: ComboOption[]) {
     let performSearch = false;
 
+    // TODO:   Why in none of the tests is temporarySearchPredicate.chapters ever set?
```
Comment (En-8) [inline user-interface/src/search/SearchScreen.tsx:150]: now or later?

File: user-interface/src/lib/components/cams/linkUtils.test.tsx [hunk 9]
```diff
@@ -4,13 +4,13 @@ import { render } from '@testing-library/react';
 describe('Test link utilities', () => {
   test('executeLinkClick should execute onClick on the given link', async () => {
     const testFunction = vi.fn();
-    const { container } = render(
+    render(
```
Comment (jamesobrooks) [inline user-interface/src/lib/components/cams/linkUtils.test.tsx:7]: @btposey, weren't you saying we should be going more to the left here?

File: user-interface/src/lib/testing/testing-utilities.ts (hunk 7 above)
Comment (jamesobrooks) [inline user-interface/src/lib/testing/testing-utilities.ts:114]: Remove

File: user-interface/src/search/SearchScreen.tsx (hunk 8 above)
Comment (btposey) [inline user-interface/src/search/SearchScreen.tsx:150]: ???

File: user-interface/src/search/SearchScreen.tsx [hunk 10]
```diff
@@ -243,13 +262,13 @@ export default function SearchScreen() {
...
                   aria-live="off"
                   onClose={handleChapterSelection}
-                  onPillSelection={handleChapterSelection}
                   onUpdateSelection={handleChapterClear}
                   onFocus={handleFilterFormElementFocus}
                   options={chapterList}
                   required={false}
                   multiSelect={true}
                   ref={chapterSelectionRef}
+                  pluralLabel={'chapters'}
```
Comment (btposey) [inline user-interface/src/search/SearchScreen.tsx:271]:
```suggestion
                  singularLabel="chapter"
                  pluralLabel="chapters"
```

**Review by btposey — CHANGES_REQUESTED** [review]

File: user-interface/src/lib/components/cams/linkUtils.test.tsx (hunk 9 above)
Comment (btposey) [reply user-interface/src/lib/components/cams/linkUtils.test.tsx:7]: I'm not entirely sure what the `container` is. I suspect it is a DOM? I was thinking about returning `React.ReactObject` refs, but not all our components take refs. So, could just be on an as needed basis since we write custom `renderWithProps` functions for each test anyhow.

File: user-interface/src/admin/privileged-identity/PrivilegedIdentity.tsx (hunk 1 above)
Comment (btposey) [reply user-interface/src/admin/privileged-identity/PrivilegedIdentity.tsx:117]: It is not.

File: user-interface/src/data-verification/transfer/SuggestedTransferCases.tsx (hunk 6 above)
Comment (jamesobrooks) [reply user-interface/src/data-verification/transfer/SuggestedTransferCases.tsx:156]: It was not easily verified. Leaving for now.

**Review by btposey — APPROVED** [review]

**Review by jamesobrooks — APPROVED** [review]
